lib/thrift/binary/framed/protocol_handler.ex

```diff
@@ -11,12 +11,13 @@ defmodule Thrift.Binary.Framed.ProtocolHandler do
   }
   require Logger
 
-  @spec start_link(reference, pid, module, {module, module}) :: GenServer.on_start
+  @spec start_link(reference, pid, module, {module, module, Keyword.t}) :: GenServer.on_start
```

Comment:
Have a look at what I just pushed. I think it's a reasonable balance between the semi-generic nature of the ProtocolHandler and the fact that it's implementation is `:ranch_tcp`-specific right now.